lib/model/message_list.dart

```diff
@@ -86,6 +88,67 @@ class MessageListView extends ChangeNotifier {
     notifyListeners();
   }
 
+  applyChangesToMessage(UpdateMessageEvent event, Message message) {
+    if (event.renderingOnly != null && event.renderingOnly == true) {
+      //TODO update inline preview
+      return;
+    }
+
+    if (!event.flags.equals(message.flags)) {
+      //TODO Possibly a new @alert on this message. Update alerts.
```

Comment:
Making the same assumption here.

Reply:
> increment, say, a badge somewhere

(This is indeed something we'll eventually want. But that'll be its own data structure somewhere, not part of MessageListView. In particular there can be anywhere from zero to many MessageListViews at a given time, for varying subsets of the messages, corresponding to the different message-list screens the user may or may not have open on the navigation stack; whereas we'll want exactly one of whatever data structure tracks things like the number of unread @-mentions the user has, or the number they have per stream etc., to drive such badges.)